### ISSUE TITLE
forbid with in with

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1031]
# Rule request

## Thesis

Let's forbid `with` statement inside another one `with` statement:

```python
# bad
with c1:
  with c2:
    ...

# good
with c1, c2:
  ...

# good when you have a lot of context
with ExitStack() as stack:
  stack.register(s1)
  stack.register(s2)
  ...
```

It's ok when with isn't the only thing in context, i.e. they have a different scope:

```python
# ok
with c1:
  with c2:
    ...
  ...

# ok
with c1:
  ...
  with c2:
    ...
```

## Reasoning

The same as in #1030. Let's fight unnecessary indentation.

**Comment by sobolevn** [comment]
We already have the opposite rule. We prefer nested `with` over `,` 
Why? Because it can be really hard to read. I've seen some shit.
